### PR TITLE
SNOW-1256899 Add structured types docs

### PR DESCRIPTION
**Pull request by sfc-gh-pfus**
### Description

SNOW-1256899 Added docs of structured types usage.

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: doc.go [hunk 1]
```diff
@@ -412,8 +414,74 @@ data types. The columns are:
...
 
+    [6] Arrays and objects can be either semistructured or structured, see more info in section below.
+
 Note: SQL NULL values are converted to Golang nil values, and vice-versa.
 
+## Semistructured and structured types
+
+Snowflake supports two flavours of "structured data" - semistructured and structured.
+Semistructured types are variants, objects and arrays without schema.
+These types are represented by strings and should be interpreted on the client side.
```
Comment (sfc-gh-dheyman-1) [inline doc.go:425]: I think this makes it more clear `When data is fetched, it's represented as strings and the client is responsible for its interpretation.`

File: doc.go [hunk 2]
```diff
@@ -412,8 +414,74 @@ data types. The columns are:
...
+Example table definition:
+
+	CREATE TABLE semistructured (v VARIANT, o OBJECT, a ARRAY)
+
+In all rows of the table values may be slightly different.
```
Comment (sfc-gh-dheyman-1) [inline doc.go:430]: `The data not have any corresponding schema, so values in table may be slightly different. `

File: doc.go [hunk 3]
```diff
@@ -412,8 +414,74 @@ data types. The columns are:
...
+3. Use it in regular scan:
+
+	var res simpleObject
+	err := rows.Scan(&res)
+
+See StructuredObject for all available operations including null support, embedding nested structs, etc.
+
+#### Retrieving structured arrays
+
+Retrieving array of simple types works exactly the same like normal values - using scan.
```
Comment (sfc-gh-dheyman-1) [inline doc.go:472]: `(...) using Scan method.`

File: doc.go (hunk 1 above)
Comment (sfc-gh-pfus) [reply doc.go]: Done.

File: doc.go (hunk 2 above)
Comment (sfc-gh-pfus) [reply doc.go]: Done.

File: doc.go (hunk 3 above)
Comment (sfc-gh-pfus) [reply doc.go]: Done.

**Pull request merged**